code/modules/power/smes_construction.dm

```diff
@@ -84,6 +84,9 @@
 /obj/machinery/power/smes/buildable/Destroy()
 	qdel(wires)
 	wires = null
+	for(var/obj/machinery/power/terminal/T in terminals)
+		T.master = null
+	terminals = null
```

Comment:
I believe this needs to move up to parent type (`/obj/machinery/power/smes`) otherwise batteryracks won't clear their terminal's `master` when destroyed.
![image](https://user-images.githubusercontent.com/14110581/80021582-b5577000-84a8-11ea-840c-8860618f1ac4.png)